### ISSUE TITLE
[ENHANCEMENT] Add additional info for Route Connectivity "Inactive" Status

### ISSUE THREAD

**Issue by rrajendran17** [harvester/harvester#6671]
**Is your enhancement related to a problem? Please describe.**

customer topology:
VM enp1s0 10.18.82.162/22(vlan 423)---------vm-network-b0------------10.18.80.1 vlan.423 EXTERNAL SWITCH vlan.420 10.18.16.1----------mgmt.420-bo-----------mgmt-br.420 10.18.17.56/22

Problem :
1.Customer created a vlan network in vlan 423 with manual mode subnet 10.18.80.0/22 and gw 10.18.80.1.
2.The "Route Connectivity" status shows as "Inactive" for the vlan vm network 423.
3.Ping from vm vlan 423 enp1s0 interface 10.18.82.162/22 towards gw 10.18.80.1 is successful.

Reason for "Route Connectivity" status "Inactive":

Route Connectivity status for a vlan vm network is decided based on "DHCP server providing the subnet/gw addr in case of Auto mode" and  ping result from harvester host towards the gw ip provided by manual or auto mode.

Ping from mgmt network towards gw 10.18.80.1 is unsuccessful in this case.
mgmt network is vlan 420 and 10.18.80.1 is in vlan 423.
Intervlan routing must be configured at the external switches for communication between different vlans.

Customer has not enabled/configured inter vlan routing, hence the ping fails.
Currently customer does not want to enable this and I believe this is acceptable in cases where customer
wants to completely isolate vlans and does not require inter vlan communication.
So Route connectivity status "Inactive" is expected in this case, but I  user should be made aware
of the reason for this and this could be expected in certain cases like above.

With just status "Inactive" user thinks that there is some underlying problem with network connectivity though that is not the case.

**Describe the solution you'd like**
Additional info in UI about status "Inactive" - harvester host unable to reach the gw <gw ip> ,ignore in cases of mgmt network
isolated from vm vlan networks.And Route connectivity should also be more self explanatory
(or) link to the page of "Route Connectivity" of vm vlan networks

**Describe alternatives you've considered**
Backend can also define some new states for connectivity like "gw unreachable" in case of ping failures and in case of mgmt network in custom vlan and gw is in another vlan, the state could be "gw isolated" etc.

**Additional context**

**Comment by starbops** [comment]
I agree. We should avoid using "Inactive" because it's inaccurate. Replacing it with more self-explanatory terms like you proposed (and perhaps with some tooltips and links to the documentation) will reduce confusion. The "Route Connectivity" check only implies the availability of other features like load balancers, not representing the status of the VM network itself.

**Comment by ParadoxGuitarist** [comment]
Hi, 

I actually opened a support ticket because we had some networks that were showing up as `Inactive` and others that were ok.  Our use case is probably different from what engineering imagined, but ultimately highlighted a potential incomplete test. I'm basing the quotes for how things work based off of that was sent to us in the support ticket and this [Docs #611 PR](https://github.com/harvester/docs/pull/611) 

From the PR: 
> For example, the [Harvester cloud provider](../rancher/cloud-provider.md) that is running in the guest cluster must access the underlying Harvester and Kubernetes APIs [...]

Similar from Support Ticket (but not in the PR): 

>Guest clusters running on the “inactive” VM network cannot access the Harvester cluster. The harvester-cloud-provider is the primary affected as also the harvester-csi-driver. It needs to interact with the kube-apiserver in the Harvester cluster. 

So in our setup, the nodes are publicly available. Some of VM Networks also get public IPs, or the gateway has a public IP. These always show as active. 

Some of the other networks that are tagged and brought in on a trunked link all pass the other other tests. However, since the VLANs are private, their gateway is in the same private VLAN. Network can't traverse VLANs, but the downstream cluster can reach the Harvester nodes.  (I've got clusters using the harvester-csi-driver in the that seem to be fully functional on "inactive" networks)

Hopefully you can clarify, but it seems like this might be a bad test case. Joining new nodes to the cluster uses this test: https://harvestercluster.example.edu/ping But it sounds like we're using pings from outside the network to try to verify connectivity outside in, which isn't the actual requirement from what's stated in the docs, which only talks about the downstream clusters needing to talk to the harvester cluster.

**Comment by rrajendran17** [comment]
@ParadoxGuitarist Can you also elaborate on what you mean by "(I've got clusters using the harvester-csi-driver in the that seem to be fully functional on "inactive" networks)" ?

I am not able to access this test link https://harvestercluster.example.edu/ping, Can you also post the steps you are trying to verify in this test case ?

Currently we require our harvester cloud provider which runs in the VM Network should be reachable from the harvester mgmt network as it has to fetch the VM metadata and other LB information from the harvester API.So in order to make sure of this connectivity, harvester network controller ping from harvester mgmt network towards the GW IP addr of the VM Network provided by DHCP server during VM Network creation.
Also FYI, https://github.com/harvester/harvester/issues/3037 This is something we are working on so that harvester cloud provider in VM Network does not need direct access to harvester mgmt network in future.

Can you elaborate on this statement "But it sounds like we're using pings from outside the network to try to verify connectivity outside in" and where exactly harvester is trying this ?

**Comment by ParadoxGuitarist** [comment]
>  Can you also elaborate on what you mean by "(I've got clusters using the harvester-csi-driver in the that seem to be fully functional on "inactive" networks)" ?

So I have clusters in rancher that show up as healthy and seem to be functioning fine that are deployed on Harvester in networks that are currently listed as "inactive". 

![Image](https://github.com/user-attachments/assets/eebb495b-33cf-48a6-9852-b7c3414cc101)

> I am not able to access this test link https://harvestercluster.example.edu/ping, Can you also post the steps you are trying to verify in this test case ?

When you join a harvester node to an existing cluster, it asks for the Management URL via the VIP or FQDN of the node as part of the installation process from the ISO.  In the case of  https://harvestercluster.example.edu/ping  I had edited the fqdn so the whole world doesn't know about our public instance. :) I think I found you on the Rancher slack, so I can DM you our real addresses there. 

> Currently we require our harvester cloud provider which runs in the VM Network should be reachable from the harvester mgmt network **as it has to fetch the VM metadata and other LB information from the harvester API.** 

Right, so that's part of my inside -> Out  vs Outside -> In comment. So As I understand it, the downstream cluster/VM/Provider has to be able to reach the Harvester API.  So from an "inside network" to get out to wherever harvester is. 

I'll use some very generic ideas to kinda point out how we're set up , Let's say I have three nodes... They're all configured the same: 
-  Our cluster/network topology looks something like this: 
   - Cluster is available at 1.1.1.1 and is available at harvestercluster.example.edu
   - node1 has a 1.1.1.11 address
   - node2 has a 1.1.1.12 address
   - node3 has a 1.1.1.13 address
   - haproxy1 is a box at 1.1.1.8 (with a private IP of 192.168.0.8)
   - haproxy2 is a box at 1.1.1.9 (with a private IP of 192.168.0.9)
- eth0 - Untagged VLAN 5 (this is the management network) it's the 1.1.1.1/16 subnet
- eth1 - Tagged VLANs 10 11 12 which are all private networks.
   - VLAN 10 - 192.168.0.0/24 Gateway is 192.168.0.1 and is available through HAProxy on `haproxy1` and `haproxy2` which also do DHCP. 
   - VLAN 11 - 10.10.10.0/24
   - VLAN 12 - 10.12.0.0/24

So in this setup harvester launches some VMs on VLAN 10. When harvester runs the test for if the `Route Connectivity` is `Active` or not, it tries to find the gateway via DHCP  by putting in that DHCP request from inside the VLAN10 VMNetwork. DHCP, which is running on haproxy1, responds and says the gateway is at `192.168.0.1`.  

Harvester responds by checking the gateway, not from inside VLAN 10, but from VLAN 5. There's no route to that inside private network, so it marks it as `Inactive`. **However**, the important part is that "[... sic] the Provider, which runs in the VM Nwork, [...] has to fetch the VM metadata and other LB information from the harvester API."

> Can you elaborate on this statement "But it sounds like we're using pings from outside the network to try to verify connectivity outside in" and where exactly harvester is trying this ?

So instead of checking to see if the network can get VM metadata and other LB information, Harvester is running a check to see if the VLANs are able to be transversed but from outside. Harvester, in this case, has a public IP and is publicly available. Private (inside) networks can reach out to it just fine. They can get that VM metadata and kubeapi access just fine. Harvester, however, can't reach inside the private network. 

My comment about the  `/ping` URI was that if that's the requirement for having Harvester nodes join as new nodes, why not ping that same URL/URI from where the DHCP req is made as a test of if the network is `Active` or `Inactive` because based off of "fetch the VM metadata and other LB information from the harvester API" that's the direction of network traffic that's important, not the harvester nodes being able to ping the vlan gateway. 


Hopefully that all makes sense, but if you need more elaboration, let me know. :)

**Comment by rrajendran17** [comment]
@ParadoxGuitarist I think I understand your topology and concern now.
I believe the way "Route Connectivity Check" in Harvester is designed is not based on a particular usecase in mind (example harvester cloud provider needs VMs metadata) but rather a generic method where we want to make sure connectivity works between harvester mgmt and VM network at the beginning of VM Network creation as VMs will be all allocated IPs in the subnet range provided by the DHCP server from that vlan.But this model fails in the following scenarios where,

case a.All VLAN network do not require connectivity to the mgmt network.
case b.In your case the VM network is able to reach the Harvester Public IP through NAT, while the GW IP received from DHCP server by Harvester mgmt network is a private GW IP, so the ping fails and eventually the status shows as "Inactive"

For case a, The term "Inactive" could be misleading where user has no usecase for  connectivity between harvester mgmt network and  the VM network and the network is healthy.

For case b, which is a case of Public and private IPs and where pinging to the private GW IP provided by DHCP server will be incorrect as it will fail though connectivity between Harvester mgmt network and VM network is fine.

Apart from this we also use "Route Connectivity Status" check for VM load balancers to make sure the VMs networks are reachable from Harvester mgmt network.I believe even "Inactive" status in this case is possible if we use private IPs for the VMs and public IP for Harvester mgmt network.

So apart from revising the status from "Inactive" to something meaningful and documentation, we need to also revisit the code in harvester-network-controller on how we confirm connectivity status as "active" which will provide false result in case of public/private IP scenario as quoted above.

cc @w13915984028 @starbops @mingshuoqiu Please add/correct if I am missing something.Thanks

**Comment by ParadoxGuitarist** [comment]
I'll add this as well: 

The terminology "Route Connectivity" might be confusing. From what I've seen in community slack, and in a few tickets I think many users are under a assumption that this is a check for a default route ( Like what'd you'd see from `ip r` ) and that it's working/responding. Personally, I thought that's what it was for and thought it was useful.  The current name is tough because a **"route"** is a specific thing, but that's not really what this check is about. 

You can take this idea or leave it, but here's what I'd suggest after hearing about what's actually getting checked, and hearing the why of it: 

- **_Replace this field with "Network Connectivity"_**  -  At least in the UI. I think you can keep the Route Connectivity bit available through the API with the same values for backwards compatibility , but have it be deprecated. This more generic term should help be more intuitive. (Or call it something else that is more intuitive) 
- Add the following new statuses: 
  - **Isolated** - There's no connectivity to the harvester cluster, or to the outside internet. I'd call this air-gapped, but that normally means something different and there's no layer 2 connection. 
  - **ICI** - Intenet Capable Isolated (Honestly I don't know what to call this, maybe `Firewalled` ) Where you can get out to the internet but can't get to the harvester cluster.
  - **Private**  (Alternatively `NATed`) - It's on a private network, but can reach the harvester cluster, but traffic isn't routable from outside the VLAN/VM Network to inside. 
  - **Full** (Alternatively: `Peered` or `Peer2Peer` The peer2peer has other connotations and might be confusing) - Traffic from inside the network can reach the Cluster (verified by a successful `/ping` uri against the cluster vip) and traffic from the Management network can reach inside. I think a good check for this is instead of pinging the gateway on the Network, try getting a DHCP address (or setting one manually in the settings like how the gateway can be in the current check) and getting a ping from one of the nodes on the Management Network. Another test might be to capture a packet sent to the broadcast address of the VM Network. (A wol packet verified with a mac or some other type of packet). Either one of those might be a better check than just pinging the gateway. You could even try all the test and let any of them pass)
  - **DHCP Failed** - Same as before and is considered a failed state. 

As I understand it the Private and Full could be green badges, and the others yellow badges with DHCP Failed being red. Anything Green gets an `Active` and everything else gets `Inactive` in the now deprecated (or not) Route Connectivity, which is still available via the API.  But since it doesn't show up in the UI, and there's less confusion for us users. I would guess that more status like this would help both Suse Support and the community with troubleshooting as it would be more apparent as to how networks might be designed (or at least configured).

**Comment by rrajendran17** [comment]
Thanks @ParadoxGuitarist for the suggestions.
It is a good point to do a check from one of the VLAN Network IP address towards the mgmt node IP, but still that will not solve the case of public/private IP,I believe we need to separate ping failure along with identifying the private IP range to categorize the VLAN Network as Private.Again it is done within the node, so we cannot be sure of there are any issues at the external switch.

The base of "Route connectivity check" is to tell user about the connectivity of VLAN Network towards the harvester mgmt network.
Not sure if we want to add more granularity to it, IMO we could keep Ping failed,DHCP Failed (Inactive) and Private,Full (Active).

Can you clarify on capturing the packet by "sending ping to broadcast address of VM Network"  ? Since we do not have any active devices, there will be no response.I do not get the exact intention behind this testcase.

Also I agree we need to change the term "Route Connectivity".

**Comment by ParadoxGuitarist** [comment]
> Can you clarify on capturing the packet by "sending ping to broadcast address of VM Network" ? Since we do not have any active devices, there will be no response.I do not get the exact intention behind this testcase.

I haven't dug into the code, but as I understand it, there technically is an active "device" with the the pod that runs the dhcp request for the VM Network/VLAN. Is that correct? The way our network is set up on some of these VLANs the DHCP request to find the gateway can't  succeed from the Management Network.  

If the goal is to: "tell user about the connectivity of VLAN Network towards the harvester mgmt network." Then why not have that same pod (or similar) run the ping test to the harvester nodes instead of having the harvester nodes reach out to the VLAN network gateway?  This setup/pod/device is also where you could monitor for a broadcast  or ping packet that originates from the harvester nodes.

**Comment by rrajendran17** [comment]
@ParadoxGuitarist to clarify, In the existing code,we run a dhcp client on the VM Network to get the gateway IP from the dhcp server and this does not represent an active device which can respond to pings.But creating one such and pinging from the device towards the mgmt network is still possible.We need to understand the pros and cons of it and decide on the solution.

**Comment by w13915984028** [comment]
As already discussed above, different VM network may have different usage and isolation requirements.
https://docs.harvesterhci.io/v1.4/networking/harvester-network#vlan-network

The enhancement could be:

Allow user to toggle `ping gateway`, `test dhcp server`, `ping Host cluster VIP (in Rancher downstream cluster case) ` on each VM network separately;  and display the status of them separately on each VM network.

It is hard to have a common summary status on the VM network list page, we may remove the status column.

**Comment by ParadoxGuitarist** [comment]
Outright removing it, as a customer, would make me a little sad. 

It's nice to have some sort of validation that the config you put in is working. I think that's why the "Route Connectivity" was so confusing as several of us customers made comments like "But the VMs can get to google, so the route works... "

I think the misconception there on our part shows the potential value. 

Outside of that, having read the (currently unpublished) docs, having the status show up in the UI to see if LB will work or not is super beneficial when it means I don't have to try to spin up test VMs and just mess around and find out the hard way.

**Comment by ihcsim** [comment]
@ParadoxGuitarist I think I get your inside-out vs. outside-in thinking. Without getting too tangled up with terminology, conventionally, a heartbeat ping flows in the inside-out direction (i.e. "i'm still here/i can reach you"), versus a routing ping which flows from a outside-in direction (i.e. "are you still there/can i reach you"). I think there are rooms and benefits for both. Feels like different users wanna see different things. Per @w13915984028, we will iterate on it for now, to see what makes sense. Appreciate the feedback.

**Comment by starbops** [comment]
I guess the name "Route Connectivity" emphasizes that the target VM network must be routable from the node's management network from the LB function's POV. That's why the network controller tries to ping the target VM network's gateway IP from the management IP address instead of pinging the management network's gateway from the transient pod that runs the DHCP client.

For example, the following VM network is deemed "Active". The network controller can ping `192.168.0.254` from `10.0.0.10`. This is exactly the same VM-type LB traffic flows.
```
 ┌───────────────────────────────────────────┐ 
 │                   ┌────────────────────┐  │ 
 │  harvester node   │ guest vm           │  │ 
 │                   │                    │  │ 
 │                   │   192.168.0.100/24 │  │ 
 │                   │          ┌──────┐  │  │ 
 │                   │          │eth0  │  │  │ 
 │                   └──────────┴───┬──┴──┘  │ 
 │                                  │        │ 
 │     ┌────────────┐      ┌────────┴───┐    │ 
 │     │10.0.0.10/24│      │            │    │ 
 │     │   mgmt-br  │      │   vm-br    │    │ 
 │     │            │      │            │    │ 
 │     └─────┬──────┘      └─────┬──────┘    │ 
 │           │                   │           │ 
 │           │                   │           │ 
 │    ┌──────┴────────┐     ┌────┴──────┐    │ 
 │    │ mgmt-bo/eth0  │     │ vm-bo/eth1│    │ 
 └────┴────┬──────────┴─────┴─────┬─────┴────┘ 
           │                      │            
           │                      │            
           │                      │            
           │                      │            
           │                      │            
           │                      │            
 ┌─────────┴───────────┐          │            
 │  10.0.0.254         │          │            
 │                     │          │            
 │        192.168.0.254├──────────┘            
 │                     │                       
 │router/l3-switch     │                       
 └─────────────────────┘                       
```

Another example:
```
 ┌───────────────────────────────────────────┐   
 │                   ┌────────────────────┐  │   
 │  harvester node   │ guest vm           │  │   
 │                   │                    │  │   
 │                   │   192.168.0.100/24 │  │   
 │                   │          ┌──────┐  │  │   
 │                   │          │eth0  │  │  │   
 │                   └──────────┴───┬──┴──┘  │   
 │                                  │        │   
 │     ┌────────────┐      ┌────────┴───┐    │   
 │     │10.0.0.10/24│      │            │    │   
 │     │   mgmt-br  │      │   vm-br    │    │   
 │     │            │      │            │    │   
 │     └─────┬──────┘      └─────┬──────┘    │   
 │           │                   │           │   
 │           │                   │           │   
 │    ┌──────┴────────┐     ┌────┴──────┐    │   
 │    │ mgmt-bo/eth0  │     │ vm-bo/eth1│    │   
 └────┴────┬──────────┴─────┴─────┬─────┴────┘   
           │                      │              
           │                      │              
 ┌─────────┴───────┐         ┌────┴────────────┐ 
 │  10.0.0.254     │         │  192.168.0.254  │ 
 │ router/l3-switch│         │                 │ 
 │  172.19.0.2     │         │ router/l3-switch│ 
 └─────────┬───────┘         │      (nat)      │ 
           │                 │   172.19.0.6    │ 
           │                 └──────┬──────────┘ 
           │                        │            
         ┌─┴────────────────────────┴──┐         
         │172.19.0.1/30  172.19.0.5/30 │         
         │                             │         
         │          router             │         
         └─────────────────────────────┘                                                       
```
This is similar to @ParadoxGuitarist's scenario. Though it's not public IP, the idea is the same. The network controller won't be able to ping `192.168.0.254` from `10.0.0.10` because it's behind an NAT. However, the guest VM can still reach the node IP or even the Internet. The LB will not work in such a situation because the target VM network is not routable from the management network. You'll see a red "Inactive" label show on the dashboard for that VM network.

---
IIRC, the "Route Connectivity" is just an informative heads-up but not used anywhere to prevent you from doing anything.

**Comment by ParadoxGuitarist** [comment]
Probably at risk of repeating myself but... 

> I guess the name "Route Connectivity" emphasizes that the target VM network must be routable from the node's management network from the LB function's POV.

As an admin, and not a network person, that's really not what it emphasizes to me. It's a list of the VM Networks so if I see "Route" I think of a route available to that VM's network, ie: `ip r` and not "Load Balancer Connectivity" or "LB Capable" . 

I get that's 100% how it was designed, and what's it's checking, but it's not clear or intuitive based off my experience. 

>IIRC, the "Route Connectivity" is just an informative heads-up but not used anywhere to prevent you from doing anything.

You're 100% correct on this as far as I can tell! 

It's just seeing a list of all my VM Networks (not on the load balancer page) then a *Big Red Route Inactive* generally throws me and it's not a far fetch to think: Something is wrong!  The network Route isn't Working! There's multiple ways the wording/phrasing/checks could be altered to provide better clarity to what exactly is being checked. 

tldr it's a great idea, but bad implementation imho.

**Comment by danielbjornadal** [comment]
​I am experiencing an issue that appears to be related to this enhancement discussion. In my Harvester cluster, the management network is configured as `1.1.1.0/24` on `VLAN 10`, with a gateway at `1.1.1.1`. I've also added a virtual machine network with the subnet `193.50.8.128/27` on `VLAN 20`, which has a gateway at `193.50.8.129`. Both networks are connected to a Cisco ASA that restricts traffic flow between them, preventing access from the management network to the virtual machine network's gateway.​

As a result, the **Route Connectivity** status for `VLAN 20` is displayed as **Inactive**, and load balancers are non-functional on this VLAN. From my understanding, this behavior aligns with the issues discussed in this thread, where the **Route Connectivity** status is determined based on the Harvester host's ability to ping the gateway IP provided during VM network creation.​

**I have a few questions regarding this situation:​**

1. Is this issue addressed in the upcoming v1.5.0 or 1.6.0 release? I noticed that this enhancement is slated for the v1.6.0 milestone. Can you confirm if the changes in this release will resolve the "Inactive" status and restore load balancer functionality in scenarios like mine, where inter-VLAN routing is restricted?​
2. Are there recommended workarounds in the meantime? While awaiting the new release, are there any configurations or adjustments I can implement to enable load balancer functionality on VLAN 20 despite the current "Inactive" route connectivity status?​

Any guidance or recommendations would be greatly appreciated.

**Comment by rrajendran17** [comment]
@danielbjornadal 
1.Yes the observation you have is related to this enhancement,
As part of this enhancement, we are planning to add more information and self explanatory terms instead of route connectivity status so that users are aware of the real purpose of this status.This status does not have to be "Active" always.
This changes will be addressed in v1.6.0

2.Regarding the load balancers, currently Load balancers work only on mgmt network which requires connectivity must exists between the VMs and the Harvester host in mgmt network so that LB can check the health status of VMs.
But in your current case, unless you have inter vlan routing this is not possible to achieve since mgmt and VM network is in different vlans.
But Load Balancers on secondary networks is planned for v1.6.0 https://github.com/harvester/harvester/issues/7500
@starbops Please add or correct me if I am missing something,Thanks

**Comment by rrajendran17** [comment]
@a110605  @torchiaf @bk201 
We will use this task to track the UI changes to make the status of "Route connectivity" more clearer to the users.
The following has to be done in UI.
a.Change the field "Route Connectivity" to more user understanding or self explanatory field like "Network Connectivity/VM Reachability from Host/" and add tooltip to it "This status indicates the connectivity from management Host to VM network needed for proper functioning of Load Balancers and other functionalities which require VMs to access management Host".

b.If the following document changes are merged, then a link to the document can also be provided to the users in the UI.
https://github.com/harvester/docs/pull/611
@starbops @jillian-maroket can we merge this for v1.6.0, so that the reference could be added in the UI ? Thanks.

**Comment by w13915984028** [comment]
Connect https://github.com/harvester/harvester/issues/766, the `route` connectivity check topic.

@rrajendran17  

When working on 766, also noticed the related code, we might enhance from those:

(1) Add a new field like `DisableConnectivityCheck`, where user can disable the check, instead of showing `inactive`.

```
type Layer3NetworkConf struct {
	Mode         Mode         `json:"mode,omitempty"`
	CIDR         string       `json:"cidr,omitempty"`
	Gateway      string       `json:"gateway,omitempty"`
	ServerIPAddr string       `json:"serverIPAddr,omitempty"`
	Connectivity Connectivity `json:"connectivity,omitempty"`
	Outdated     bool         `json:"outdated,omitempty"`
}
```

(2) Run related check task on  new job from the `network-helper`, which also needs an refactor to have an additional network interface on the target network, and the dhcp & ping all run on the additional interface

https://github.com/harvester/network-controller-harvester/blob/4ec00f753ff2d38ca4664e4dd8d53418c7ee7266/pkg/controller/manager/nad/controller.go#L355


(3) For LB, we can update the code (validator) & doc: `VM` type LB only works on mgmt network, for second network, an independent third-party LB is needed to work around;  before Harvester officially supports `VM` type LB to work on second network.

WDYT? thanks.

**Comment by rrajendran17** [comment]
@w13915984028 Above points sounds good 1) and 3)
But I did not get the purpose of the point 2, we are already running a dhclient on the VM Network to get the gw ip and we are checking the connectivity from mgmt to this gw ip of the vm network.

**Comment by w13915984028** [comment]
Regarding the point 2), some additional info:

```

The current & legacy code adds &/ ensures this route, e.g.:

   vlan 100   |<-->|<-->    Rancher Server
              |    | 
   vlan 200   |    |<-->    Harvester(mgmt)     <---------
              |                                          |  cloud-provider-harvester
   vlan 300   |<----------->VM for guest cluster  --------
   
   
   Inter-vlan connectivity already set: vlan 100 <--> 200,  100 <--> 300
   
   Due to the restriction that the Harvester provided guest cluster has `cloud-provider-harvester` feature, 
   which currently needs to connectivity with Harvester mgmt network
   It requries additional
             vlan 200 <--> 300
    
   This breaks some security rules.
   
   
   Potential solutions:
   
   (1) Set an endpoint on Rancher Server to serve the cloud-provider-harvester related request
   (2) Like CSI, use unix socket to shortcurt
   
   (3) Rethink the `cloud-provider-harvester`, it talks to Harvester just in case the guest cluster 
   requires LB VIP resources from a POOL defined in Harvester.   for the control that: a guest cluster
  can't consume out of capacity resources
   
    In below cases, the connectivity looks not required.
    (3.1) none harvester-cloud-provider
    (3.2) harvester-cloud-provider vip dhcp mode



The `route` connectivity, is mainly for above case, but:

(1) The VM network might just serve the traditional VM workloads
(2) No  `harvester-cloud-provider vip dhcp mode` used


Then:
The connectivey check is better to be configurable:
  Disable/Enable
  Source network: mgmt or vm network
  ...
  
  
```
